src/controllers/csi/gc/images.go

```diff
@@ -0,0 +1,79 @@
+package csigc
+
+import (
+	"os"
+	"path/filepath"
+	"time"
+
+	"github.com/Dynatrace/dynatrace-operator/src/installer/image"
+	"github.com/spf13/afero"
+)
+
+var maxImageAge = time.Hour * 24 * 14 // 14 days
+
+func (gc *CSIGarbageCollector) runImageGarbageCollection() {
+	imageDirs, _ := getImageCacheDirs(gc.fs)
+	if imageDirs == nil {
+		return
+	}
+
+	imageCachesToDelete := collectUnusedImageCaches(gc.fs, imageDirs)
+
+	deleteImageCaches(gc.fs, imageCachesToDelete)
+}
+
+func getImageCacheDirs(fs afero.Fs) ([]os.FileInfo, error) {
+	imageDirs, err := afero.Afero{Fs: fs}.ReadDir(image.CacheDir)
+	if os.IsNotExist(err) {
+		log.Info("No image cache to clean up")
+		return nil, nil
+	}
+	if err != nil {
+		log.Error(err, "Failed to read image cache directory")
+		return nil, err
+	}
+	return imageDirs, err
+}
+
+func collectUnusedImageCaches(fs afero.Fs, imageDirs []os.FileInfo) []string {
+	var toDelete []string
+	for _, imageDir := range imageDirs {
+		if !imageDir.IsDir() {
+			continue
+		}
+		modificationTime, err := getRelevantModificationTime(fs, imageDir.Name())
+		if err != nil {
+			log.Error(err, "failed to get modification time of image cache", "imageDir", imageDir.Name())
+			continue
+		}
+		if time.Since(modificationTime) > maxImageAge {
+			toDelete = append(toDelete, filepath.Join(image.CacheDir, imageDir.Name()))
+		}
+	}
+	return toDelete
+}
+
+// getRelevantModificationTime returns the last modification time of an image in the cache
+// an image is not a single file but a directory of several files.
+// Most of which don't change when the image is tried to be pulled again.
+// The index.json of the image is a file that is modified in reliable way.(after every pull)
```

Comment:
```suggestion
// getRelevantModificationTime returns the last modification time of an extracted image in the cache.
// Most of the extracted files do not change after an update.
// Therefore, the index.json is checked for its modification time, as it is always updated after an image pull.
```

Reply:
AFAIK the image is not really 'extracted' in the `data/cache/<digest>`, its just everything the image comes with, the blobs are still compressed